File: src/main/java/TKBuild.java


/**
 * Created by noname on 30.11.2014.
 */
public class TKBuild {
    TKBuild(){}
    private TokenJAV.Op tok;
    private String val = "";

    public TokenJAV create()
    {
        return new TokenJAV(tok, val);
    }
    public TKBuild T(TokenJAV.Op tok)
    {
        this.tok = tok;
        return this;
    }
    public TKBuild V(String val)
    {
        this.val = val;
        return this;
    }
}

File: src/main/scala/parser/TKBuild.scala
package parser

import interpreter.TokenJAV

/**
  * Created by noname on 30.11.2014.
  */
class TKBuild() {
  private var tok: TokenJAV.Op.Value = null
  private var `val`: String = ""

  def create: TokenJAV = {
    TokenJAV(tok, `val`)
  }

  def T(tok: TokenJAV.Op.Value): TKBuild = {
    this.tok = tok
    this
  }

  def V(`val`: String): TKBuild = {
    this.`val` = `val`
    this
  }
}